src/crimson/os/seastore/random_block_manager/rbm_device.cc

```diff
@@ -18,41 +18,79 @@ namespace crimson::os::seastore::random_block_device {
 #include "crimson/os/seastore/logging.h"
 SET_SUBSYS(seastore_device);
 
-RBMDevice::mkfs_ret RBMDevice::do_mkfs(device_config_t config) {
-  LOG_PREFIX(RBMDevice::mkfs);
+RBMDevice::mkfs_ret RBMDevice::do_primary_mkfs(device_config_t config, int shard_num) {
+  LOG_PREFIX(RBMDevice::do_primary_mkfs);
   return stat_device(
   ).handle_error(
     mkfs_ertr::pass_further{},
     crimson::ct_error::assert_all{
-    "Invalid error stat_device in RBMDevice::mkfs"}
-  ).safe_then([this, FNAME, config=std::move(config)](auto st) {
+    "Invalid error stat_device in RBMDevice::do_primary_mkfs"}
+  ).safe_then([this, FNAME, config=std::move(config), shard_num=shard_num](auto st) {
     super.block_size = st.block_size;
     super.size = st.size;
     super.feature |= RBM_BITMAP_BLOCK_CRC;
     super.config = std::move(config);
-    assert(super.journal_size);
+    assert(super.journal_size > 0);
     assert(super.size >= super.journal_size);
+    assert(shard_num > 0);
+
+    std::vector<rbm_shard_info_t> shard_infos(shard_num);
+    for (int i = 0; i < shard_num; i++) {
+      uint64_t aligned_size = 
+	(super.size / shard_num) -
+	((super.size / shard_num) % super.block_size);
+      shard_infos[i].size = aligned_size;
+      shard_infos[i].start_offset = i * aligned_size;
+      shard_infos[i].journal_size = super.journal_size;
+      assert(shard_infos[i].size > shard_infos[i].journal_size);
+    }
+    super.shard_infos = shard_infos;
+    super.shard_num = seastar::smp::count;
```

Comment:
here ,should be super.shard_num = shard_num.  since for EphemeralRBMDevice test it is 1.